legend-engine-xt-relationalStore-bigquery-execution-tests/src/test/java/org/finos/legend/engine/plan/execution/stores/relational/connection/test/ExternalIntegration_TestConnectionAcquisitionWithFlowProvider_BigQuery.java

```diff
@@ -146,41 +128,45 @@ public void testBigQueryGCPWIFConnection_subject() throws Exception
         Connection connection = this.connectionManagerSelector.getDatabaseConnection((MutableList<CommonProfile>) null, systemUnderTest);
-        testConnection(connection, "select * from `legend-integration-testing.integration_dataset1.table1`");
+        testConnection(connection, 1,"select * from `legend-integration-testing.integration_dataset1.table1`");
     }
 
     @Test
     public void testBigQueryGCPWIFConnection_profile() throws Exception
     {
         RelationalDatabaseConnection systemUnderTest = this.bigQueryWithGCPWIFSpec();
         Connection connection = this.connectionManagerSelector.getDatabaseConnection((MutableList<CommonProfile>) null, systemUnderTest);
-        testConnection(connection, "select * from `legend-integration-testing.integration_dataset1.table1`");
+        testConnection(connection, 1,"select * from `legend-integration-testing.integration_dataset1.table1`");
     }
 
     private RelationalDatabaseConnection bigQueryWithGCPADCSpec() throws Exception
     {
-        BigQueryDatasourceSpecification bigQueryDatasourceSpecification = new BigQueryDatasourceSpecification();
-        bigQueryDatasourceSpecification.projectId = "legend-integration-testing";
-        bigQueryDatasourceSpecification.defaultDataset = "integration_dataset1";
-        GCPApplicationDefaultCredentialsAuthenticationStrategy authSpec = new GCPApplicationDefaultCredentialsAuthenticationStrategy();
-        return new RelationalDatabaseConnection(bigQueryDatasourceSpecification, authSpec, DatabaseType.BigQuery);
+        RelationalConnectionTest rt = new RelationalConnectionTest();
+        RelationalDatabaseConnection rdb = rt.readRelationalConnections(
+                rt.getResourceAsString("/org/finos/legend/engine/server/test/bigqueryRelationalDatabaseConnections.json"))
+                .stream()
+                .filter(relationalDatabaseConnection -> relationalDatabaseConnection.authenticationStrategy instanceof GCPApplicationDefaultCredentialsAuthenticationStrategy)
+                .collect(Collectors.toList())
+                .get(0);
+        return rdb;
     }
 
-    private RelationalDatabaseConnection bigQueryWithGCPWIFSpec()
-    {
-        BigQueryDatasourceSpecification bigQueryDatasourceSpecification = new BigQueryDatasourceSpecification();
-        bigQueryDatasourceSpecification.projectId = "legend-integration-testing";
-        bigQueryDatasourceSpecification.defaultDataset = "integration_dataset1";
-        GCPWorkloadIdentityFederationAuthenticationStrategy authSpec = new GCPWorkloadIdentityFederationAuthenticationStrategy();
-        authSpec.serviceAccountEmail = "integration-bq-sa1@legend-integration-testing.iam.gserviceaccount.com";
-        return new RelationalDatabaseConnection(bigQueryDatasourceSpecification, authSpec, DatabaseType.BigQuery);
+    private RelationalDatabaseConnection bigQueryWithGCPWIFSpec() throws JsonProcessingException {
```

Comment:
If you keep different element values for each connection instead of empty string, we can use getRelationalConnectionByElement to select the appropriate one for test

Reply:
changed